### PR TITLE
Implement Continuous Mode jumpToMeasure

### PR DESCRIPTION
**Pull request by bluebarryz**

### CODE REVIEW

File: fe/lib/main.dart [hunk 1]
```diff
@@ -20,19 +24,294 @@ class MyApp extends StatelessWidget {
...
+              future: _svgs,
+              builder:
+                  (BuildContext context, AsyncSnapshot<SvgPicture> snapshot) {
+                if (snapshot.hasData && orientation == Orientation.landscape) {
+                  return Scaffold(
+                    body: ListView(
+                        controller: _scrollController,
+                        scrollDirection: Axis.vertical,
+                        children: [snapshot.data!]),
+                    floatingActionButton: FloatingActionButton(
```
Comment (max-y-huang) [inline fe/lib/main.dart:292]: We could have a TODO comment for eventually removing the button.

File: fe/lib/main.dart [hunk 2]
```diff
@@ -20,19 +24,294 @@ class MyApp extends StatelessWidget {
...
+        .toList();
+    final stafflineElementsCount = stafflineElements.length;
+    final linesPerGroup = svgXml
+        .findAllElements("g")
+        .where((line) => (line.getAttribute('class')!.contains('vf-measure') &&
+            line.getAttribute('id') == '1'))
+        .length;
+
+    for (var i = 0; i < stafflineElementsCount; i += linesPerGroup) {
+      double maxY = -1;
```
Comment (max-y-huang) [inline fe/lib/main.dart:231]: Same boundary-checking idea as before.

**Review by max-y-huang — APPROVED** [review]

File: fe/lib/main.dart [hunk 3]
```diff
@@ -20,19 +24,294 @@ class MyApp extends StatelessWidget {
...
+}
+
+class GroupInfo {
+  int startingMeasure;
+  double minY;
+
+  GroupInfo(this.startingMeasure, this.minY);
+}
+
+class MinMaxYCoords {
```
[the remark this answers is not in the document]
Comment (bluebarryz) [reply fe/lib/main.dart:46]: Dart's Range class is only for ints. Couldn't find anything similar for `double`

File: fe/lib/main.dart [hunk 4]
```diff
@@ -20,19 +24,302 @@ class MyApp extends StatelessWidget {
...
+  //  <g class="vf-measure" id="1">
+  bool _isMeasure(XmlElement elem) {
+    if (elem.getAttribute('class') == null) {
+      return false;
+    } else if (elem.getAttribute('class')!.contains('vf-measure') &&
+        elem.getAttribute('id') != '-1') {
+      return true;
+    } else {
+      return false;
+    }
```
Comment (ngtr6788) [inline fe/lib/main.dart:119]: Just my personal opinion, something to make the code simpler
```suggestion
    final classAttribute = elem.getAttribute('class');
    return classAttribute != null 
      && classAttribute!.contains('vf-measure')
      && elem.getAttribute('id') != '-1';
```

File: fe/lib/main.dart [hunk 5]
```diff
@@ -20,19 +24,302 @@ class MyApp extends StatelessWidget {
...
+  bool _isLine(XmlElement elem) {
+    final String? dAttribute = elem.getAttribute("d");
+    if (elem.name.local == "path" &&
+        dAttribute != null &&
+        elem.getAttribute("class") == null &&
+        lineCoordinatesRegex.hasMatch(dAttribute)) {
+      return true;
+    } else {
+      return false;
+    }
```
Comment (ngtr6788) [inline fe/lib/main.dart:133]:
```suggestion
    return elem.name.local == "path" &&
        dAttribute != null &&
        elem.getAttribute("class") == null &&
        lineCoordinatesRegex.hasMatch(dAttribute);
```

File: fe/lib/main.dart [hunk 6]
```diff
@@ -20,19 +24,302 @@ class MyApp extends StatelessWidget {
...
+        group1Info.startingMeasure.compareTo(group2Info.startingMeasure));
+    final int veryFirstMeasureId = groups[0].startingMeasure;
+    for (var groupInfo in groups) {
+      groupInfo.startingMeasure -= veryFirstMeasureId;
+    }
+  }
+
+  void jumpToMeasure(int measureNumber) {
+    int groupNumber = _getGroupForMeasure(measureNumber);
+    final double offset = _height * 1 / 8;
```
Comment (ngtr6788) [inline fe/lib/main.dart:260]: Perhaps you can put the 1/8 as some constant elsewhere?

**Review by ngtr6788 — APPROVED** [review]

**Pull request merged**